client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -164,12 +164,17 @@ const siteSetupFlow: Flow = {
 					if ( ! siteSlug ) {
 						return;
 					}
+
+					// If user clicks "Skip to Dashboard" on goals step,
+					// Default site_intent to "build"
+					const updatedSiteIntent = intent ? intent : SiteIntent.Build;
```

Comment:
Note this isnt specifically tied to this dashboard button on the goals step, so im curious if there are any other areas this change could potentially bleed into?

That said, testing around and escaping out of other flows early I see the intent the same as expected. So it seems to be looking good in my testing.